src/shared/lib/encryption/encryption.ts

```diff
@@ -0,0 +1,110 @@
+import { Buffer } from 'buffer';
+
+import { IV_LENGTH } from '@shared/lib';
+
+import { crypto } from './crypto';
+
+type GetPrivateKeyProps = { userId: string; email: string; password: string };
+type GetPublicKeyProps = {
+  privateKey: number[];
+  appletPrime: number[];
+  appletBase: number[];
+};
+
+type GetAESKeyProps = {
+  userPrivateKey: number[];
+  appletPublicKey: number[];
+  appletPrime: number[];
+  appletBase: number[];
+};
+
+type EncryptDataProps = { text: string; key: number[] };
+type DecryptDataProps = { text: string; key: string };
+
+class Encryption {
```

Comment:
Please come up with a better name. Encryption means a process and cannot be interpreted as a class. EncryptionService or EncryptionManager, for example